### ISSUE TITLE
Field names associated with ReflectiveTypeAdapterFactory errors.

### ISSUE THREAD

**Issue by aerb** [google/gson#788]
Would it be possible to include the failed field name in exceptions thrown from ReflectiveTypeAdapterFactory? This would be for debugging purposes. Right now, if there is a type mismatch during a read for example, a wrapped IllegalStateException gets thrown with the expected and actual types. It would be useful to also include the field this failure occurred at, as the message of the JsonSyntaxException.

Looking at ReflectiveTypeAdapterFactory.java this shouldn't be too difficult.

**Comment by swankjesse** [comment]
We typically include the JSON path of the offending field. Is that insufficient? Mind pasting a representative stack trace?

**Comment by aerb** [comment]
Hmmm ok, here's a stacktrace I'm getting if I trigger a type mismatch. Is there a debugging flag that needs to be set?
```
com.google.gson.JsonSyntaxException: java.lang.IllegalStateException: Expected BOOLEAN but was NUMBER
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:221)
 at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40)
 at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
 at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:40)
 at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82)
 at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
 at com.google.gson.Gson.fromJson(Gson.java:861)
 at com.google.gson.Gson.fromJson(Gson.java:926)
 at com.google.gson.Gson.fromJson(Gson.java:899)
 at com.eblock.auction.json.JsonMappings.as(JsonMappings.java:132)
 at com.eblock.auction.async.LiveTasks$1.inBg(LiveTasks.java:44)
 at com.eblock.auction.async.TryTask$1.call(TryTask.java:57)
 at com.eblock.auction.async.TryTask$1.call(TryTask.java:54)
 at java.util.concurrent.FutureTask.run(FutureTask.java:237)
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
 at java.lang.Thread.run(Thread.java:818)
Caused by: java.lang.IllegalStateException: Expected BOOLEAN but was NUMBER
 at com.google.gson.internal.bind.JsonTreeReader.expect(JsonTreeReader.java:139)
 at com.google.gson.internal.bind.JsonTreeReader.nextBoolean(JsonTreeReader.java:160)
 at com.google.gson.internal.bind.TypeAdapters$3.read(TypeAdapters.java:161)
 at com.google.gson.internal.bind.TypeAdapters$3.read(TypeAdapters.java:151)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:117)
 at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:217)
 ... 20 more
```

**Comment by JakeWharton** [comment]
Are you using Gson 2.4 or newer?

**Comment by aerb** [comment]
Sorry, should have included that. Yeah 2.6.

**Comment by JakeWharton** [comment]
Ok. It can be added to the exception then. Most others are already showing
it.

**Comment by aerb** [comment]
Cool, thanks.

**Comment by Marcono1234** [comment]
Looks like this was fixed by #871.
